config/minimalist-templates/popups_templates.yaml

```
@@ -0,0 +1,1396 @@
+              icon: "mdi:chart-line"
+              type: "custom:button-card"
+              template:
+                - "popup_button"
+                - "popup_button_stats"
+              tap_action:
+                action: "fire-dom-event"
+                browser_mod:
+                  command: "popup"
+                  hide_header: true
+                  card:
+                    type: "custom:button-card"
+                    template: "popup_power_outlet_stats"
+                    variables:
+                      ulm_popup_power_outlet_entity: "[[[ return variables.ulm_popup_power_outlet_entity ]]]"
+                      ulm_popup_power_outlet_sensor1: "[[[ return variables.ulm_popup_power_outlet_sensor1 ]]]"
+                      ulm_popup_power_outlet_sensor2: "[[[ return variables.ulm_popup_power_outlet_sensor2 ]]]"
+                      ulm_popup_power_outlet_graph_sensor: "[[[ return variables.ulm_popup_power_outlet_graph_sensor ]]]"
+          item2:
+            card:
+              icon: "mdi:history"
+              type: "custom:button-card"
+              template:
+                - "popup_button"
+                - "popup_button_selected"
+                - "popup_button_history"
+
+popup_light:
+  hold_action:
+    action: "fire-dom-event"
+    browser_mod:
+      command: "popup"
+      large: true
+      hide_header: true
+      card:
+        type: "custom:button-card"
+        template: "popup_light_brightness"
+        variables:
+          ulm_popup_light_entity: "[[[ return entity.entity_id ]]]"
+
+popup_light_brightness:
+  styles:
+    grid:
+      - grid-template-areas: >
+          [[[
+                var headers = [];
+                var contents = [];
+                var footers = [];
+                var brightness_modes = ["brightness", "hs", "color_temp", "rgb", "rgbw"];
+                var color_temp_modes = ["color_temp"];
+                var color_modes = ["hs", "rgb", "rgbw", "xy"];
+                if (states[variables.ulm_popup_light_entity].attributes.supported_color_modes.some(r => brightness_modes.includes(r))) {
+                    headers.push("header");
+                    contents.push("content1");
+                    footers.push("footer");
+                }
+                if (states[variables.ulm_popup_light_entity].attributes.supported_color_modes.some(r => color_temp_modes.includes(r))) {
+                    headers.push("header");
+                    contents.push("content2");
+                    footers.push("footer");
+                }
+                if (states[variables.ulm_popup_light_entity].attributes.supported_color_modes.some(r => color_modes.includes(r))) {
+                    headers.push("header");
+                    contents.push("content3");
+                    footers.push("footer");
+                }
+                return "\"" + headers.join(" ") + "\" " + "\"" + contents.join(" ") + "\" " + "\"" + footers.join(" ") + "\"";
```

```suggestion
                return "\"" + headers.join(" ") + "\" " + "\"" + contents.join(" ") + "\" " + "\"" + footers.join(" ") + "\"";
                return "\"" + "header ".repeat(contents.length) + "\" " + "\"" + contents.join(" ") + "\" " + "\"" + "footer ".repeat(contents.length) + "\"";
```
☝️ Small improvement. Instead of filling 3 lists, you can just count the amount of headers you need to fill 🙂

---

Thank you !!